File: app/src/main/java/info/nightscout/androidaps/plugins/general/food/CarbService.java
package info.nightscout.androidaps.plugins.general.food;

import java.util.List;

import info.nightscout.androidaps.MainApp;
import info.nightscout.androidaps.R;
import info.nightscout.androidaps.db.CareportalEvent;
import info.nightscout.androidaps.interfaces.Constraint;
import info.nightscout.androidaps.plugins.general.nsclient.NSUpload;
import info.nightscout.androidaps.plugins.treatments.CarbsGenerator;
import info.nightscout.androidaps.utils.SafeParse;

import static info.nightscout.androidaps.utils.DateUtil.now;

public class CarbService {

    public CarbService() {}

    static int calculateCarb(List<Food> foodList) {
        int carbs = 0;
        for (Food food : foodList) {
            carbs += food.carbs * food.portionCount;
        }
        return carbs;
    }
}

File: app/src/main/java/info/nightscout/androidaps/plugins/general/food/CarbService.kt
package info.nightscout.androidaps.plugins.general.food

import info.nightscout.androidaps.MainApp
import info.nightscout.androidaps.R
import info.nightscout.androidaps.db.CareportalEvent
import info.nightscout.androidaps.interfaces.Constraint
import info.nightscout.androidaps.plugins.general.food.FoodUtils.Companion.roundDoubleToInt
import info.nightscout.androidaps.plugins.general.nsclient.NSUpload
import info.nightscout.androidaps.plugins.treatments.CarbsGenerator
import info.nightscout.androidaps.utils.SafeParse

import info.nightscout.androidaps.utils.DateUtil.now
import kotlin.math.round

class CarbService {
    companion object {

        fun calculateCarb(foodList: List<Food>): Int {
            var carbs = 0.0
            for (food in foodList) {
                carbs += food.carbs * food.portionCount
            }
            return roundDoubleToInt(carbs)
        }
    }
}
